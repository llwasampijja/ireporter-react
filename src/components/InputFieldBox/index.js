// react libraries
import React, { Fragment } from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const InputFieldBox = (props) => {
  const {
    inputFieldName,
    inputFieldValue,
    inputFieldId,
    inputFieldType,
    inputFieldClass,
    inputFieldOnchange,
    inputFieldPlaceholder,
    children,
  } = props;

  return (
    <Fragment>
      <input
        id={inputFieldId}
        name={inputFieldName}
        value={inputFieldValue}
        type={inputFieldType}
        className={inputFieldClass}
        onChange={inputFieldOnchange}
        placeholder={inputFieldPlaceholder}
      />
      {children}
    </Fragment>
  );
};

InputFieldBox.propTypes = {
  inputFieldName: PropTypes.string,
  inputFieldValue: PropTypes.string,
  inputFieldId: PropTypes.string,
  inputFieldType: PropTypes.string,
  inputFieldClass: PropTypes.string,
  inputFieldOnchange: PropTypes.func,
  inputFieldPlaceholder: PropTypes.string,
  children: PropTypes.node,
};

InputFieldBox.defaultProps = {
  inputFieldName: '',
  inputFieldValue: '',
  inputFieldId: '',
  inputFieldType: '',
  inputFieldClass: '',
  inputFieldOnchange: () => {},
  inputFieldPlaceholder: '',
  children: '',
};

export default InputFieldBox;
