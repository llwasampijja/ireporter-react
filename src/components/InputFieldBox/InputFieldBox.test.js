// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// component
import InputFieldBox from '.';

describe('tests the inputfieldbox component', () => {
  const wrapper = shallow(<InputFieldBox />);
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
