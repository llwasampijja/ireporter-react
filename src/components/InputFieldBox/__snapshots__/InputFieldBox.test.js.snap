// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the inputfieldbox component should match snapshop 1`] = `
<Fragment>
  <input
    className=""
    id=""
    name=""
    onChange={[Function]}
    placeholder=""
    type=""
    value=""
  />
</Fragment>
`;
